src/go/rdctl/cmd/snapshot.go

```diff
@@ -63,38 +63,46 @@ func exitWithJsonOrErrorCondition(e error) error {
 // If the main process is running, stops the backend, calls the
 // passed function, and restarts the backend. If it cannot connect
 // to the main process, just calls the passed function.
-func wrapSnapshotOperation(wrappedFunction cobraFunc) cobraFunc {
-	return func(cmd *cobra.Command, args []string) error {
-		appPaths, err := paths.GetPaths()
-		if err != nil {
-			return fmt.Errorf("failed to get paths: %w", err)
-		}
-		if err := createBackendLock(appPaths.AppHome); err != nil {
-			return err
-		}
-		defer removeBackendLock(appPaths.AppHome)
-		if err := ensureBackendStopped(cmd); err != nil {
-			return err
-		}
-		if err := wrappedFunction(cmd, args); err != nil {
+func wrapSnapshotOperation(cmd *cobra.Command, appPaths paths.Paths, resetOnFailure bool, wrappedFunction cobraFunc) error {
+	if err := createBackendLock(appPaths.AppHome); err != nil {
+		return err
+	}
+	defer removeBackendLock(appPaths.AppHome)
+	if err := ensureBackendStopped(cmd); err != nil {
+		return err
+	}
+	if err := wrappedFunction(); err != nil {
+		if resetOnFailure {
 			factoryreset.DeleteData(appPaths, true)
-			return err
 		}
-		// Note that this does not wait for the backend to be in the
-		// STARTED (or DISABLED if k8s is disabled) state. This allows
-		// removeBackendLock() to be called as a deferred function while
-		// keeping the state of the backend lock file in sync with the
-		// main process backendIsLocked variable.
-		return ensureBackendStarted()
+		return err
+	}
+	// Note that this does not wait for the backend to be in the
+	// STARTED (or DISABLED if k8s is disabled) state. This allows
+	// removeBackendLock() to be called as a deferred function while
+	// keeping the state of the backend lock file in sync with the
+	// main process backendIsLocked variable.
+	return ensureBackendStarted()
+}
+
+func getConnectionInfo() (*config.ConnectionInfo, error) {
```

Comment:
As I said in https://github.com/rancher-sandbox/rancher-desktop/pull/5715#discussion_r1364556070, I think we should keep the `config.GetConnectionInfo()` error handling code the same rather than adding `getConnectionInfo()` back.

Reply:
The failing tests drove me to this solution. Refactoring this should be a separate issue.

Reply:
I created https://github.com/rancher-sandbox/rancher-desktop/issues/5792 to track this.